src/ports/server/component.ts

```diff
@@ -668,6 +707,105 @@ export async function createServerComponent({
       return res.sendStatus(200)
     })
 
+    // Store identity endpoint with signed fetch middleware validation
+    app.post('/identities', signedFetchMiddleware({ optional: false }), async (req: Request, res: Response) => {
+      try {
+        // Extract AuthIdentity from request body
+        const { identity }: { identity: AuthIdentity } = req.body
+
+        if (!identity) {
+          return sendResponse<InvalidResponseMessage>(res, 400, {
+            error: 'AuthIdentity is required in request body'
+          })
+        }
+
+        // Validate auth chain using the same logic as /requests endpoint
+        try {
+          const { sender: identitySender, finalAuthority } = await validateAuthChain(identity.authChain)
+
+          // Verify that the ephemeral wallet address matches the finalAuthority from auth chain
+          if (identity.ephemeralIdentity.address.toLowerCase() !== finalAuthority.toLowerCase()) {
+            return sendResponse<InvalidResponseMessage>(res, 403, {
+              error: 'Ephemeral wallet address does not match auth chain final authority'
+            })
+          }
+
```

Comment:
There's another check that we're missing to be 100% safe about the input, we need to check that the received address belongs to the received private key.
We can do that by instantiating the wallet with the private key and checking it.
`new ethers.Wallet(identity.ephemeralIdentity.privateKey).address = identity.ephemeralIdentity.address`